edb/server/cluster.py

```diff
@@ -216,69 +218,59 @@ async def _read_server_status(stream: asyncio.StreamReader):
-                    )
-                except asyncio.TimeoutError:
-                    raise ClusterError(
-                        f'EdgeDB server did not initialize '
-                        f'within {timeout} seconds'
-                    ) from None
-
-                self._effective_port = data['port']
-                stat_writer.close()
-                left -= (time.monotonic() - started)
+            except asyncio.TimeoutError:
+                raise ClusterError(
+                    f'EdgeDB server did not initialize '
+                    f'within {timeout} seconds'
+                ) from None
 
-            while True:
-                started = time.monotonic()
-                try:
-                    conn = await edgedb.async_connect(
-                        host=str(self._runstate_dir),
-                        port=self._effective_port,
-                        database=edgedb_defines.EDGEDB_SUPERUSER_DB,
-                        user=edgedb_defines.EDGEDB_SUPERUSER,
-                        timeout=left,
-                    )
-                except (OSError, asyncio.TimeoutError,
-                        edgedb.ClientConnectionError):
-                    left -= (time.monotonic() - started)
-                    if left > 0.05:
-                        await asyncio.sleep(0.05)
-                        left -= 0.05
-                        continue
-                    raise ClusterError(
-                        f'could not connect to edgedb-server '
-                        f'within {timeout} seconds') from None
-                except edgedb.AuthenticationError:
-                    return
-                else:
-                    await conn.aclose()
-                    return
-
-        asyncio.run(test(timeout))
-
-    def _admin_query(self, query):
-        conn_args = self.get_connect_args().copy()
-        conn_args['host'] = str(self._runstate_dir)
-        conn_args['admin'] = True
-        conn_args['user'] = edgedb_defines.EDGEDB_SUPERUSER
-        conn_args['database'] = edgedb_defines.EDGEDB_SUPERUSER_DB
-        conn = self.connect(**conn_args)
+            self._effective_port = data['port']
+            self._tls_cert_file = data['tls_cert_file']
+            stat_writer.close()
 
-        try:
-            return conn.query(query)
-        finally:
-            conn.close()
+        left = timeout
+        if status_sock is not None:
+            started = time.monotonic()
+            asyncio.run(test())
+            left -= (time.monotonic() - started)
+
+        if self._admin_query("SELECT ();", f"{min(1, int(left))}s"):
+            raise ClusterError(
+                f'could not connect to edgedb-server '
+                f'within {timeout} seconds') from None
+
+    def _admin_query(self, query, wait_until_available="0s"):
+        return subprocess.call(
+            [
+                sys.executable,
+                "-m",
+                "edb.cli",
```

Comment:
I think we should call the `edgedb` command here explicitly; `edb.cli` is just a wrapper.

Reply:
But the `edgedb` command is not always in the `PATH` - finding the path to the binary is the same code as `edb.cli`(`.__main__`) I think?

Reply:
This method is only used in tests to configure test clusters, using `edb.cli` here is appropriate.

Reply:
Ohhh this is biting the nightly build test where we don't have the `edgedb` CLI binary. I'll change it to use `edb.protocol.protocol` to connect to the UNIX socket directly and run admin queries there.

Reply:
We actually have it in `PATH` there, so it might be easier to simply do use unprefixed "edgedb" here?

Reply:
Ah yes, sure thing!